modules/core/src/test/java/org/apache/ignite/internal/event/EventProducerTest.java

```diff
@@ -45,26 +42,23 @@ public void simpleAsyncTest() {
 
         CompletableFuture<?> eventHandleFuture = producer.fireEvent(TestEvent.TEST, new TestEventParameters(0L));
 
-        assertFalse(eventHandleFuture.isDone());
+        assertThat(eventHandleFuture, willTimeoutFast());
 
         future.complete(true);
 
-        assertTrue(eventHandleFuture.isDone());
+        assertThat(eventHandleFuture, willCompleteSuccessfully());
```

Comment:
And this?

Reply:
This way we will check that the future completed successfully without errors instead of two checks.

Reply:
I think the idea was that the future would complete immediately

Reply:
fix it